carta/cpp/core/ScriptedClient/ScriptFacade.cpp

```diff
@@ -768,10 +772,15 @@ QStringList ScriptFacade::getImageDimensions( const QString& controlId ) {
     if ( obj != nullptr ){
         Carta::Data::Controller* controller = dynamic_cast<Carta::Data::Controller*>(obj);
         if ( controller != nullptr ){
-            resultList = controller->getImageDimensions( );
-            if ( resultList[0] == "" ) {
+            std::vector<int> dimensions = controller->getImageDimensions();
+            if ( dimensions.size() == 1 && dimensions[0] == 0 ) {
```

Comment:
I think this is the only special case I need to check for. It is what gets returned by `Controller::getImageDimensions()` as a special error case. I can't see an empty list being returned. I think any of the other values you mentioned would only come about if an image has dimensions all equal to 0. Could such an image even exist, and if so, could it possibly be loaded without causing other problems?

Maybe I'm wrong, though. If you think I should be checking for more cases here, that can certainly be done.